src/main/java/com/swrve/segment/SwrveIntegration.java

```diff
@@ -63,8 +63,22 @@ public void identify(IdentifyPayload identify) {
   @Override
   public void track(TrackPayload track) {
     super.track(track);
-    SwrveSDK.event(track.event(), track.properties().toStringMap());
-    logger.verbose("SwrveSDK.event(%s, %s)", track.event(), track.properties().toStringMap());
+    Map<String, String> payload = new HashMap<>();
+    for (String key : track.properties().keySet()) {
+      Object value = track.properties().get(key);
+      Map<String, Object> valueMap = (value instanceof Map) ? (Map) value : null;
```

Comment:
Will they always be only 2 levels deep?

Reply:
Yes they have been told it will only support 1 level of nesting